internal/config/main/parse.go

```diff
@@ -0,0 +1,244 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"text/template"
+
+	"github.com/iancoleman/strcase"
+
+	"github.com/Masterminds/sprig/v3"
+	"github.com/forestgiant/sliceutil"
+	"github.com/pkg/errors"
+	"golang.stackrox.io/kube-linter/internal/stringutils"
+	"golang.stackrox.io/kube-linter/internal/utils"
+	"k8s.io/gengo/parser"
+	"k8s.io/gengo/types"
+)
+
+const (
+	metadataMarker   = "+"
+	viperTagKey      = "viper"
+	paramsStructName = "Config"
+)
+
+const (
+	fileTemplateStr = `// Code generated by kube-linter flag codegen. DO NOT EDIT.
+// +build !flagcodegen
+
+package config
+
+import (
+	"github.com/spf13/viper"
+	"github.com/spf13/cobra"
+)
+
+// AddFlags, walks through config.Check struct and bind its Member to Cobra command 
+// and add respective Viper flag 
+func AddFlags(c *cobra.Command, v *viper.Viper) {
+	{{- range . }}
+		
+	{{- if eq .FlagDesc.AddFlag true }}
+		{{- if eq .FlagDesc.Type "stringSlice" }}		
+		c.Flags().StringSlice("{{ .FlagDesc.Name }}", nil, "{{ .FlagDesc.Description }}")
+		{{- else if eq .FlagDesc.Type "boolean" }}
+		c.Flags().Bool("{{ .FlagDesc.Name }}", false, "{{ .FlagDesc.Description }}")
+		{{- end }}
```

Comment:
Why about the other types? If we don't support them right now that's fine, but we should throw an error so that this doesn't fail silently